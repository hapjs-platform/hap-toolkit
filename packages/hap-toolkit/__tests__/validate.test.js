/*
 * Copyright (c) 2021-present, the hapjs-platform Project Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

const validateJson = require('../gen-webpack-conf/validate').validateJson

const manifestJson = {
  package: 'com.application.demo',
  name: '{{appName}}',
  versionName: '1.0.0',
  versionCode: 1,
  minPlatformVersion: 1070,
  icon: '/Common/logo.png',
  banner: '/Common/banner.png',
  deviceTypeList: ['phone'],
  features: [{ name: 'system.prompt' }, { name: 'system.router' }, { name: 'system.shortcut' }],
  permissions: [{ origin: '*' }],
  config: {
    logLevel: 'log',
    dsl: {
      name: 'vue'
    }
  },
  router: {
    entry: 'Demo',
    pages: {
      Demo: {
        component: 'index'
      },
      DemoDetail: {
        component: 'index'
      },
      About: {
        component: 'index'
      }
    }
  },
  display: {
    titleBarBackgroundColor: '#f2f2f2',
    titleBarTextColor: '#414141',
    pages: {
      Demo: {
        titleBarText: '示例页',
        menu: false
      },
      DemoDetail: {
        titleBarText: '详情页'
      },
      About: {
        menu: false
      }
    }
  },
  menuBarData: {
    menuBar: true,
    menuBarStyle: 'light',
    shareTitle: 'this is menuBar',
    shareDescription: 'this is shareDescription'
  }
}

const wrongJson = {
  package: 123,
  versionName: '1.0.0',
  versionCode: '1',
  minPlatformVersion: 1050,
  icon: 123,
  banner: 456,
  deviceTypeList: ['phone', 123],
  features: [{ name: 'system.prompt' }, { name: 'system.router' }, { name: 123 }],
  // TODO 校验 permissions
  permissions: [{ origin: '*' }],
  config: {
    logLevel: 'not enum',
    // TODO 校验 dsl
    dsl: {
      name: 'vue'
    }
  },
  router: {
    pages: '/path/to/page'
  },
  display: {
    titleBarBackgroundColor: 123123,
    titleBarTextColor: 123123,
    pages: 'should be object'
  },
  menuBarData: {
    menuBar: 'true',
    menuBarStyle: '123',
    shareTitle: 123,
    shareDescription: 'this is shareDescription'
  }
}

it('validateJson', () => {
  let errors = validateJson(manifestJson, 'manifest.json').map((err) => err.message)
  expect(errors.join('\n\n')).toMatchSnapshot()
  errors = validateJson(wrongJson, 'manifest.json').map((err) => err.message)
  expect(errors.join('\n\n')).toMatchSnapshot()
})
