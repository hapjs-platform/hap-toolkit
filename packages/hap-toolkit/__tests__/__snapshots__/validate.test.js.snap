// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validateJson 1`] = `""`;

exports[`validateJson 2`] = `
"校验 manifest.json 错误 
package 字段必须为 string 类型
参考: https://doc.quickapp.cn/framework/manifest.html

校验 manifest.json 错误 
icon 字段必须为 string 类型
参考: https://doc.quickapp.cn/framework/manifest.html

校验 manifest.json 错误 
banner 字段必须为 string 类型
参考: https://doc.quickapp.cn/framework/manifest.html

校验 manifest.json 错误 
versionCode 字段必须为 Integer 类型
参考: https://doc.quickapp.cn/framework/manifest.html

校验 manifest.json 错误 
features.2.name 字段必须为 string 类型
参考: https://doc.quickapp.cn/framework/manifest.html

校验 manifest.json 错误 
config.logLevel 字段只能为 off, error, warn, info, log, debug
参考: https://doc.quickapp.cn/framework/manifest.html

校验 manifest.json 错误 
router.pages 字段必须为 object 类型
参考: https://doc.quickapp.cn/framework/manifest.html

校验 manifest.json 错误 
router 字段必须为 object 类型
参考: https://doc.quickapp.cn/framework/manifest.html

校验 manifest.json 错误 
display.titleBarBackgroundColor 字段必须为 string 类型
参考: https://doc.quickapp.cn/framework/manifest.html

校验 manifest.json 错误 
display.titleBarTextColor 字段必须为 string 类型
参考: https://doc.quickapp.cn/framework/manifest.html

校验 manifest.json 错误 
display.pages 字段类型必须为 'object'
参考: https://doc.quickapp.cn/framework/manifest.html

校验 manifest.json 错误 
menuBarData.menuBar 字段必须为 boolean 类型
参考: https://doc.quickapp.cn/framework/manifest.html

校验 manifest.json 错误 
menuBarData.menuBarStyle 字段只能为 adjustPan 或 adjustResize
参考: https://doc.quickapp.cn/framework/manifest.html

校验 manifest.json 错误 
menuBarData.shareTitle 字段必须为 string 类型
参考: https://doc.quickapp.cn/framework/manifest.html

校验 manifest.json 错误 
deviceTypeList.1 字段类型必须为 'string'
参考: https://doc.quickapp.cn/framework/manifest.html

校验 manifest.json 错误 
必须包含 'package'、'name'、'versionName'、'versionCode'、'minPlatformVersion'、'icon'、'pages' 字段
参考: https://doc.quickapp.cn/framework/manifest.html"
`;
